vault/seal/awskms/awskms.go

```diff
@@ -99,19 +99,16 @@ func (k *AWSKMSSeal) SetConfig(config map[string]string) (map[string]string, err
 		k.region = "us-east-1"
 	}
 
-	// Check and set AWS access key and secret key
-	k.accessKey = os.Getenv("AWS_ACCESS_KEY_ID")
-	if k.accessKey == "" {
-		if accessKey, ok := config["access_key"]; ok {
-			k.accessKey = accessKey
-		}
+	// Check and set AWS access key, secret key, and session token
+	var accessKey, secretKey, sessionToken string
+	if os.Getenv("AWS_ACCESS_KEY_ID") == "" {
```

Comment:
I think that since we use `credsConfig.GenerateCredentialChain()` below, and that includes adding an Env cred provider if the AWS key is unset, we can just do away with all fussing with pulling AWS environment variables here and let the Env provider do the right thing.

Reply:
Looks like the Env provider supports pulling specifically the access key, secret key, and session token, but nothing else.